src/python/WMCore/WorkQueue/WorkQueueBackend.py

```diff
@@ -158,23 +158,23 @@ def insertElements(self, units, parent=None):
         self.insertWMSpec(units[0]['WMSpec'])
         newUnitsInserted = []
         for unit in units:
-
             # cast to couch
             if not isinstance(unit, CouchWorkQueueElement):
                 unit = CouchWorkQueueElement(self.db, elementParams=dict(unit))
 
+            if unit._couch.documentExists(unit.id):
+                self.logger.info('Element "%s" already exists, skip insertion.' % unit.id)
+                continue
+
```

Comment:
Hi @amaltaro I seem to fail grasping the need of reordering of these checks (The one above and the `if parent` one below.) Could you please shed some light?

Reply:
I moved this check up, because when the document already exists, we do not (re-)insert it into CouchDB, neither we return this document/object back from this method. So, it looks to me the sooner we check and skip this document, the better. 
Of course, it could be a problem if this list of documents is still used later in the chain somehow, which I am not 100% sure.

Reply:
But I am going to revert it just to reduce the amount of changes required for the couch migration. Thanks Todor.

Reply:
Thanks Alan! Just to note, I was not requesting a change to it, I was asking more like... for me to understand it.